ethereum/geth_client.go

```diff
@@ -206,37 +200,33 @@ func (gc *gethClient) getGethTxOpts(accountName string) (*bind.TransactOpts, err
 	return opts, nil
 }
 
-func (gc *gethClient) queueTaskTransactionStatus(accountID identity.CentID, txHash string) (txID uuid.UUID, err error) {
-	tx, err := gc.txService.CreateTransaction(accountID, "polling Ethereum transaction status")
+// QueueTransactionStatusTask starts a new queuing transaction check task.
+func QueueTransactionStatusTask(
+	accountID identity.CentID,
+	txHash common.Hash,
+	txService transactions.Service,
+	queuer queue.TaskQueuer, updateHandler func(status transactions.Status) error) (txID uuid.UUID, err error) {
+	tx, err := txService.CreateTransaction(accountID, "Polling Ethereum transaction status")
 	if err != nil {
 		return txID, err
 	}
-	_, err = gc.queue.EnqueueJob(TransactionStatusTaskName, map[string]interface{}{
+
+	var next bool
+	if updateHandler != nil {
+		next = true
```

Comment:
Why is a next flag needed?

I understood it in a way that I can register a handler closure in the transactions and after the 
transaction status is updated the handler func will be called?

The actual task isn't aware that he might have a successor. The update transaction is taking care.
 So why the next flag?

Reply:
Is the reason because we are using the same tx.ID for multiply task to update to transactionStatus to pending instead of success?

Reply:
@manuelpolzhofer you came to the right conclusion, to track the status of a certain network action (transaction) all queued tasks for that action needs to have the same unique txID and also have a status field.